### ISSUE TITLE
Allow top level await for ES6 modules

### ISSUE THREAD

**Issue by pebly** [eslint/eslint#13663]
**Please describe what the rule should do:**
To allow await outside a function.

**What new ECMAScript feature does this rule relate to?**
As NodeJS v14 and experimental features, developers are allowed to use await top-level.

**What category of rule is this? (place an "X" next to just one item)**

- [x] Warns about a potential error (problem)
- [ ] Suggests an alternate way of doing something (suggestion)
- [ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**
Eslint already warns you about await used on top-level.
```
Parsing error: Cannot use keyword 'await' outside an async functioneslint
```
But you can't disable this warning.

**Are you willing to submit a pull request to implement this rule?**
No.

**Comment by mdjermanovic** [comment]
Hi @pebly, thanks for the issue!

The [top-level await proposal](https://github.com/tc39/proposal-top-level-await) is still in stage 3.

Per our [policy about experimental features](https://github.com/eslint/eslint#what-about-experimental-features), ESLint's default parser will support this syntax if and when it reaches stage 4.

In the meantime, you can use ESLint with an [external parser](https://eslint.org/docs/user-guide/configuring#specifying-parser) that supports this syntax, like [@babel/eslint-parser](https://github.com/babel/babel/tree/main/eslint/babel-eslint-parser) (I think you'll also need [@babel/plugin-syntax-top-level-await](https://babeljs.io/docs/en/babel-plugin-syntax-top-level-await) plugin).

**Comment by pebly** [comment]
I understand, but I don't want to install these all dependencies for just this small thing. At least, I want to disable await checking at all if needed.

Is there another solution than an external parser? Anyway, I use it with VSCode and eslint extenstion.

**Comment by mdjermanovic** [comment]
ESLint doesn't have a rule that checks for `await` at the top level, it's our default parser (`espree`) throwing an error since the `await` keyword isn't allowed there per the actual language specification.

We recommend using [@babel/eslint-parser](https://github.com/babel/babel/tree/main/eslint/babel-eslint-parser) for experimental syntax. An alternative is to fork `espree`, make changes and use your own version as an external parser, but it seems easier to use babel.
